lib/Compiler.js

```diff
@@ -381,6 +381,12 @@ class Compiler {
 	// e.g. move compilation specific info from Modules into ModuleGraph
 	_cleanupLastCompilation() {
 		if (this._lastCompilation !== undefined) {
+			for (const childCompilation of this._lastCompilation.children) {
+				for (const chunk of childCompilation.chunks) {
+					ChunkGraph.clearChunkGraphForChunk(chunk);
```

Comment:
I was initially also up modules as well:

```js
  for (const module of childCompilation.modules) {
				ChunkGraph.clearChunkGraphForModule(module);
				ModuleGraph.clearModuleGraphForModule(module);
				module.cleanupForCache();
  }
```

but it looks like `ChunkGraph.clearChunkGraphForChunk(chunk)` is the only change necessary to fix the memory leak.